src/com/google/cose/Sign1Message.java

```diff
@@ -32,21 +32,30 @@
 public class Sign1Message extends CoseMessage {
   private final byte[] message;
   private final byte[] signature;
+  private final byte[] originalProtectedHeadersData;
 
-  Sign1Message(Map protectedHeaders, Map unprotectedHeaders, byte[] message, byte[] sign) {
+  Sign1Message(
+      Map protectedHeaders,
+      byte[] originalProtectedHeadersData,
+      Map unprotectedHeaders,
+      byte[] message,
+      byte[] sign) {
     super(protectedHeaders, unprotectedHeaders);
     this.message = message;
     this.signature = sign;
+    this.originalProtectedHeadersData = originalProtectedHeadersData;
   }
 
   public static class Builder {
     private Map protectedHeaders;
+    private byte[] originalProtectedHeadersData = null;
```

Comment:
nit: `= null` is not necessary, null's the default

Reply:
Done